capstone/src/components/App/App.jsx

```diff
@@ -1,28 +1,56 @@
 import './App.css';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import axios from 'axios';
 import Visitor from '../Visitor/Visitor';
-
+import User from '../User/User';
 // CONTEXTS
-// import LoginContext from '../Contexts/LoginContext';
+import UserContext from '../../Contexts/UserContext';
 // import SignUpContext from '../Contexts/SignUpContext';
 
+export async function getUserInfo(userId) {
+  const values = { objectId: userId };
+  const baseURL = 'http://localhost:3001';
+  return axios.post(`${baseURL}/visitor/user`, values);
+}
+
 export default function App() {
   // TODO: Handle state with useContext and useMemo
-  const [isLoggedIn, setIsLoggedIn] = useState(localStorage.getItem('current_user_id') !== null);
-
+  const [isLoggedIn, setIsLoggedIn] = useState(localStorage.getItem('current_user_id') != null);
+  const [userData, setUserData] = useState({});
+  async function retrieveUserData() {
```

Comment:
Since we're declaring this inside the component anyway, does it need to be its own function? IMO it's actually a bit clearer if we inline this on line 32 so it's very obvious when this logic is run. You may need to use [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) notation.